pigskin.py

```diff
@@ -406,22 +407,72 @@ def parse_shows(self):
             episodes_data = self.make_request(episodes_url, 'get')['modules']['archive']['content']
 
             # 'season' is often left unset. It's impossible to know for sure,
-            # but the year of the broadcast date seems like a sane best guess.
-            # TODO: but apparently 'scheduleDate' often contains errors. Yay...
+            # but the year of the current Season seems like a sane best guess.
             season_list = set([episode['season'].replace('season-', '')
-                               if episode['season'] else episode['scheduleDate'][:4]
+                               if episode['season'] else current_season
                                for episode in episodes_data])
 
             show_dict[show['title']] = season_list
 
-        # RedZone
+            # Adding NFL-Network as a List of dictionary containing oher dictionaries.
+            # episode_thumbnail = {videoId, thumbnail}
+            # episode_id_dict = {episodename, episode_thumbnail{}}
+            # episode_season_dict = {episode_season, episode_id_dict{}}
+            # show_season_dict = {show_title, episode_season_dict{}}
+            # The Function returns all Season and Episodes
+            for episode in episodes_data:
+                episode_thumbnail = {}
+                episode_id_dict = {}
+                episode_season_dict = {}
+                show_season_dict = {}
+                episode_name = episode['title']
+                episode_id = episode['videoId']
+                if episode['season']:
+                    episode_season = episode['season'].replace('season-', '')
+                else:
+                    episode_season = current_season
+                # Using Episode Thumbnail if not present use theire corresponding Show Thumbnail
+                if episode['videoThumbnail']['templateUrl']:
+                    episode_thumbnail[episode_id] = episode['videoThumbnail']['templateUrl'].replace('https://nflgamepass.kleutner.de/nfl.php', self.base_url)
```

Comment:
I'm assuming you forgot to remove the ``https://nflgamepass.kleutner.de/nfl.php``

Reply:
thanks for pointing out my typo could you delete the url out of your comment?

Reply:
done. :-)